app/main.py

```diff
@@ -1,8 +1,26 @@
 class Person:
-    # write your code here
-    pass
+
+    people: [dict[str]] = {}
+
+    def __init__(self, name: str, age: str) -> None:
+        self.name = name
+        self.age = age
+        Person.people[self.name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    result_list = [
+        Person(name=person["name"], age=person["age"])
+        for person in people
+    ]
+
+    for person in people:
+        if "wife" in person and person["wife"] is not None:
```

Comment:
You can simplify conditions. use .get method on the dict.
The same with husband